akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -385,27 +378,48 @@ private[http] object HttpServerBluePrint {
                 oneHundredContinueResponsePending = true
                 r.copy(createEntity = with100ContinueTrigger(r.createEntity))
               } else r
+              lastRequestOnConnectionSummary = s"${r.method.value} ${r.uri.path}"
               push(requestPrepOut, rs)
+
             case MessageEnd ⇒
               messageEndPending = false
               push(requestPrepOut, MessageEnd)
-            case MessageStartError(status, info) ⇒ finishWithIllegalRequestError(status, info)
+
+            case MessageStartError(status, info) ⇒
+              finishWithIllegalRequestError(status, info)
+
             case x: EntityStreamError if messageEndPending && openRequests.isEmpty ⇒
               // client terminated the connection after receiving an early response to 100-continue
               completeStage()
+
             case x ⇒
               push(requestPrepOut, x)
           }
-        override def onUpstreamFinish() =
+
+        override def onUpstreamFinish(): Unit =
           if (openRequests.isEmpty) completeStage()
           else complete(requestPrepOut)
+
+        override def onUpstreamFailure(ex: Throwable): Unit = {
+          ex match {
+            case t: TcpIdleTimeoutException ⇒
```

Comment:
Is there still a reason to special case here? Or what's the reason to add logging here? Most people won't see debug login anyway, right?